src/Nancy.Swagger.Annotations/SwaggerObjects/AnnotatedOperation.cs

```diff
@@ -44,33 +44,72 @@ public AnnotatedOperation(string name, MethodInfo handler, ISwaggerModelCatalog
+            //then check for RouteParamAttributes on themethod parameters
+            CreateSwaggerParametersFromParameters(handler, paramsList);
 
-            Parameters = CreateSwaggerParameters(infos);
+            Parameters = paramsList;
 
         }
 
-        private IEnumerable<Parameter> CreateSwaggerParameters(IEnumerable<ParameterInfo> infos)
+        private void CreateSwaggerParametersFromMethodAttributes(MethodInfo info, List<Parameter> paramsList)
         {
-            var result = new List<Parameter>();
-            foreach (var info in infos)
+            var routeParamsAttributes = info.GetCustomAttributes<RouteParamAttribute>(true);
+            foreach (var attrib in routeParamsAttributes)
             {
-                var paramAttrs = info.GetCustomAttributes<RouteParamAttribute>();
+                if (string.IsNullOrEmpty(attrib.Name))
+                    throw new ArgumentNullException("Name", "RouteParam name cannot be null when used on method.");
+                //we can only have 1 body parameter
+                if (attrib.ParamIn == ParameterIn.Body)
+                {
+                    //if we already have a body parameter then ignore any subsequent ones.
+                    if (paramsList.Where(x => x.GetType() == typeof(AnnotatedBodyParameter)).Any())
+                    {
+                        continue;
+                    }
+                    if ((attrib.ParamType == null))
+                        throw new ArgumentNullException("ParamType", "ParamType for Body must bespecified when RouteParam used on method.");
+
+                    paramsList.Add(new AnnotatedBodyParameter(attrib.Name, attrib.ParamType, attrib, _modelCatalog));
+                }
+                //ignore duplicate named parameters.
+                if (!paramsList.Any(x => x.Name.Equals(attrib.Name)))
+                {
+                    paramsList.Add(new AnnotatedParameter(attrib.Name, attrib.ParamType, attrib));
+                }
+            }
+        }
+
+        private void CreateSwaggerParametersFromParameters(MethodInfo info, List<Parameter> paramsList)
+        {
+            //get parameters that have RouteParamAttrbutes 
+            var infos = info.GetParameters().Where(x => x.GetCustomAttributes<RouteParamAttribute>().Any()).ToList();
+
+            foreach (var paramInfo in infos)
+            {
+                var paramAttrs = paramInfo.GetCustomAttributes<RouteParamAttribute>(true);
                 //Body param trumps all other attributes.
                 var bodyParamAttr = paramAttrs.FirstOrDefault(x => x.ParamIn == ParameterIn.Body);
                 if (bodyParamAttr != null)
                 {
-                    result.Add(new AnnotatedBodyParameter(info, _modelCatalog));
+                    //Attribute on method parameter will replace one specified on the method.
+                    var existingBodyParam = paramsList.FirstOrDefault(x => x.GetType() == typeof(AnnotatedBodyParameter));
```

Comment:
Why would there ever be two?

I feel like an exception should be thrown, rather than just ignore one. That way the use can easily know they misconfigured something.

Reply:
There shouldn't be two, but it's always possible a developer could accidently specify two. Changed to throw an exception.